docs/examples/writing_drivers/Creating-Instrument-Drivers.ipynb

```diff
@@ -83,8 +112,34 @@
     "  \n",
     "Golden rule: if a `Parameter` is settable, it must always accept its own output as input.\n",
     "\n",
-    "In most cases you will probably be adding parameters via the `add_parameter` method on the instrument class as shown in the example below. \n",
+    "### Adding parameters to intruments"
+   ]
+  },
+  {
+   "cell_type": "raw",
+   "metadata": {
+    "raw_mimetype": "text/restructuredtext"
+   },
+   "source": [
+    "QCoDeS provides two ways of adding parameters to intruments:\n",
+    "\n",
+    "- (Conventional style) Via the :meth:`qcodes.instrument.base.Instrument.add_parameter` method.\n",
+    "- (Decorator style) Via the :func:`@qcodes.instrument.base.add_parameter <qcodes.instrument.base.add_parameter>` decorator function.\n",
+    "\n",
+    "The more recent decorator style provides two key advantages:\n",
+    "\n",
+    "- Allows to override parameters when inheriting them from another :class:`qcodes.instrument.base.Instrument` class.\n",
```

Comment:
strictly speaking the conventional style does allow one to override parameters, it's just that the to-be-overridden parameter needs to be removed first from ``self.parameters``, so i suggest to make it clear in this sentence, and also refer to the "Dynamically adding and removing parameters" that explain that:
```suggestion
    "- Allows to override parameters when inheriting them from another :class:`qcodes.instrument.base.Instrument` class without the need for interacting with the instrument's dictionary of parameters (``self.parameters``), see [Dynamically adding and removing parameters section](Dynamically-adding-and-removing-parameters).\n",
```